.github/workflows/run-framework-tests.yaml

```diff
@@ -0,0 +1,53 @@
+on:
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - main
+
+name: run-framework-tests
+
+jobs:
+  run-framework-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Nest
+        uses: actions/checkout@v4
+        with:
+          path: ''
+          fetch-depth: 0 # Checkout everything to get access to the tags
+          repository: ${{ github.repository }}
+          ref: ${{github.event.pull_request.head.ref}}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout Flock
+        uses: actions/checkout@v4
+        with:
+          path: 'flock'
+          fetch-depth: 0 # Checkout everything to get access to the tags
+          repository: 'flutter/flutter'
+          ref: 'master'
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      #- name: Apply Nest patches to Flock
+      #  run: |
+      #    tools/git-import-patches patches
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Setup Android Tools
+        run: |
+          sudo apt-get install android-tools-adb
+
+      - name: Run Tests
+        run: |
+          PATH="$PATH:`pwd`/flock/bin"
```

Comment:
What are the things you're trying to add to the path? It might be good to explicitly add the individual tools to the path that you need so that future maintainers aren't forced to figure that out.

Reply:
this is just adding the flutter cli into the path